File: 02 Binary/Reverse Bits/ReverseBits.java
import java.util.Scanner;

class Solution {
    public int reverseBits(int n) {
        int result = 0;
        for (int i = 0; i < 32; i++) {
            result += n & 1;
            n >>>= 1;
            if (i < 31)
                result <<= 1;
        }
        return result;
    }
}

public class ReverseBits {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            int num = sc.nextInt();
            Solution obj = new Solution();
            System.out.println(obj.reverseBits(num));
            t--;
        }
        sc.close();
    }
}

File: 02 Binary/Reverse Bits/ReverseBits.go
package main

import "fmt"

func reverseBits(n int) int {
	var result int
	for i := 0; i < 32; i++ {
		result |= (n & 1) << (31 - i)
		n >>= 1
	}
	return result
}

func main() {
	var t int
	fmt.Scan(&t)
	for t > 0 {
		t--
		var n int
		fmt.Scan(&n)
		fmt.Println(reverseBits(n))
	}
}
